src/pip/_internal/vcs/helix_core.py

```diff
@@ -0,0 +1,286 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import random
+import re
+import socket
+import string
+
+from pip._vendor.six.moves.urllib import parse as urllib_parse
+from pip._internal.vcs import VersionControl, vcs
+
+logger = logging.getLogger(__name__)
+
+client_spec_template = """
+Owner:   {user}
+Host:    {host}
+Client:  {name}
+Root:    {client_path}
+View:    /{server_path}/... //{name}/...
+Options: allwrite rmdir
+"""
+
+
+def split_url(url, environ=os.environ):
```

Comment:
Default of `None` is usually better, using the `if environ is None: ...` pattern.

Reply:
Can you explain more why it's better? I think I'm missing something.

Reply:
Two reasons: it's more idiomatic, and it's also friendlier to callers when composing with other functions. To get the default behavior, you can pass None instead of having to duplicate the default value in all calling functions.